diagnostic_aggregator/src/aggregator.cpp

```diff
@@ -96,9 +99,9 @@ Aggregator::Aggregator()
   other_analyzer_ = std::make_unique<OtherAnalyzer>(other_as_errors);
   other_analyzer_->init(base_path_);  // This always returns true
 
-  diag_sub_ = n_->create_subscription<DiagnosticArray>(
-    "/diagnostics", rclcpp::SystemDefaultsQoS(), std::bind(&Aggregator::diagCallback, this, _1));
-  agg_pub_ = n_->create_publisher<DiagnosticArray>("/diagnostics_agg", 1);
+  diag_sub_ = n_->create_subscription<diagnostic_msgs::msg::DiagnosticArray>(
+    "/diagnostics", rclcpp::SystemDefaultsQoS().keep_last(history_depth_), std::bind(&Aggregator::diagCallback, this, _1));
+  agg_pub_ = n_->create_publisher<diagnostic_msgs::msg::DiagnosticArray>("/diagnostics_agg", 1);
```

Comment:
To highlight the actual changes:

```suggestion
  diag_sub_ = n_->create_subscription<DiagnosticArray>(
    "/diagnostics", rclcpp::SystemDefaultsQoS().keep_last(history_depth_), std::bind(&Aggregator::diagCallback, this, _1));
  agg_pub_ = n_->create_publisher<DiagnosticArray>("/diagnostics_agg", 1);
```